### PR TITLE
[ISSUE 109] Back-end Repo Setup

### LINKED ISSUE #109: [Task]: Backend Repo Setup
### Summary

Set up the backend portion of the repository with the language decided in #38 and the set of code quality tools selected in #101 

### User Stories

- As an **open source contributor**, I want to:
  - be able to reference documentation explaining how to use the developer tools, so that I don't have to learn how to use these tools on my own in order to contribute to the project
  - have full test coverage for the codebase, so that I know when I've introduce code that changes or breaks existing behavior
  - have code formatting and standards enforced with automated tooling, so that I don't have to learn and check that my code adheres to those standards manually
- As a **maintainer of the project**, I want to:
  - have code quality checks run automatically on each push, so that formatting, linting, or security issues are caught before being deployed to production
  - ensure that new contributions meet certain thresholds for test coverage, so that code contributions from internal and external developers also include tests which validate the code's behavior

### Acceptance Criteria

- [x] An `api/` sub-directory has been created with a `README` that includes instructions for how to install and run the code in this sub-directory
- [x] At least 2 dependencies have been added to the project, and the dependency manager selected in #101 automatically checks for dependency conflicts and offers suggestions for how to reconcile those conflicts:
- [x] A dummy test has been created (i.e. `assert True`) that users can execute to verify they've installed the project correctly
- [x] The following checks ~~are run before every commit~~ are possible:
  - [x] Auto-formatting
  - [x] Linting (basic)
  - [x] Type checks (optional)
  - [ ] ~~Commit formatting (preferably conventional commit)~~
- [x] The following checks are run with each push to the remote GitHub repo, and these checks are run against all minor versions of the backend language we plan to support:
  - [x] Auto-formatting
  - [x] Linting (complete)
  - [x] Type checks
  - [ ] ~~Dependency vulnerability checks~~
  - [ ] ~~Open source license checks~~
  - [x] Unit tests
  - [x] Test coverage
- [ ] A PR is blocked from merging if any of the checks above fail
- [ ] At least 2 people who did not contribute to this PR have installed and run the dummy test locally

### Proposed Testing Strategy

1. **Unit tests:** Create a `test_sample` test that simply asserts `True` or asserts the version number
2. **Manual Tests:** Someone who did not contribute to the PR checks out the PR locally, follows the instructions in the README to install the code, run the tests and the code quality checks locally, and all checks/tests pass.

### PR DESCRIPTION
**Pull request by daphnegold**
## Summary
Fixes #109 

## Changes proposed
This adds a base Flask application with tooling, tests and goodies.

## Context for reviewers
Please pull down the branch, run through the `getting-started.md` to check that everything works, and run the tests!

**Fast follow issues:**

Pre-commit hooks #223 
Vulnerability checks #239 
License checks #238 

Note: After this PR is on main, will need someone with admin privileges to add GA checks passing as a blocker for merging PRs.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,7 @@ downloads/
 eggs/
 .eggs/
 lib/
```
Comment (daphnegold) [inline .gitignore:24]: I didn't feel confident removing this from the `.gitignore` so I just included another rule below to include our testing `/lib` but let me know!

**Comment by SammySteiner** [comment]
I was able to run tests successfully:
image

**Review by SammySteiner — APPROVED** [review]
Looks great to me! 
I was able to start the app and access localhost:8080/docs as well as run tests successfully.

**Review by widal001 — APPROVED** [review]
This looks great, such a major achievement 🎉 👏 !!!

I saw the code quality checks are running, which was the primary ask for this ticket, and I was able to get the API spun up locally (see screenshot below).

### Request
My one suggestion, is that we may want to re-evaluate where some of the docs are stored and make it easier for folks to discover the steps needed to install and spin up the API from the main README. But let's create a follow-up ticket for that work to unblock this sizable PR from being merged.

Screenshot 2023-07-14 at 5 08 15 PM

**Pull request merged**